apps/judicial-system/web/src/routes/Prosecutor/Indictments/Defendant/PoliceCaseList/PoliceCaseList.tsx

```diff
@@ -0,0 +1,498 @@
+
+    const policeCaseNumbers =
+      workingCase.origin === CaseOrigin.LOKE
+        ? // If the case is a LÖKE case, we never change the first police case number
+          [first, ...rest.sort(compare)]
+        : unsortedPoliceCaseNumbers.sort(compare)
+
+    return { policeCaseNumbers, indictmentSubtypes, crimeScenes }
+  }
+
+  const handleUpdateIndictmentCounts = (
+    old: WorkingCaseUpdate,
+    updated: WorkingCaseUpdate,
+    updates: IndexedPoliceCaseUpdate[],
+  ) => {
+    if (!workingCase.indictmentCounts) {
+      // No indictment counts to update
+      return
+    }
+
+    const { policeCaseNumbers: oldPoliceCaseNumbers } = old
+    const {
+      indictmentSubtypes: updatedIndictmentSubtypes,
+      crimeScenes: updatedCrimeScenes,
+    } = updated
+
+    let updatedIndictmentCounts = [...workingCase.indictmentCounts]
+
+    for (const {
+      index,
+      update: {
+        policeCaseNumber: policeCaseNumberUpdate,
+        subtypes: subtypesUpdate,
+        crimeScene: crimeSceneUpdate,
+      },
+    } of updates) {
+      // Assumptions:
+      // 1. At most one update per indictment count
+      // 2. For each update, only one of the updates is set
+      updatedIndictmentCounts = updatedIndictmentCounts.map(
+        (indictmentCount) => {
+          const policeCaseNumber = oldPoliceCaseNumbers[index]
+
+          if (indictmentCount.policeCaseNumber !== policeCaseNumber) {
+            return indictmentCount
+          }
+
+          if (policeCaseNumberUpdate) {
+            // This case is handled by the server, but we need to update the local state
+            return {
+              ...indictmentCount,
+              policeCaseNumber: policeCaseNumberUpdate,
+            }
+          }
+
+          const policeCaseNumberSubtypes =
+            subtypesUpdate ?? updatedIndictmentSubtypes[policeCaseNumber]
+
+          if (policeCaseNumberSubtypes.length === 0) {
+            // If there are no subtypes, we don't update the indictment count
+            return indictmentCount
+          }
+
+          // handle changes based on police case subtype changes
+          const indictmentCountSubtypes =
+            indictmentCount.indictmentCountSubtypes || []
+          const updatedIndictmentCountSubtypes = indictmentCountSubtypes.filter(
+            (subtype) => policeCaseNumberSubtypes.includes(subtype),
+          )
+          const updatedIndictmentCount = {
+            ...indictmentCount,
+            indictmentCountSubtypes: updatedIndictmentCountSubtypes,
+            policeCaseNumberSubtypes,
+          }
+
+          const incidentDescription = getIncidentDescription(
+            updatedIndictmentCount,
+            gender,
+            crimeSceneUpdate ?? updatedCrimeScenes[policeCaseNumber],
+            formatMessage,
+            { [policeCaseNumber]: policeCaseNumberSubtypes },
+          )
+
+          // TODO: Handle all updates, except indictment desctiption server side
```

Comment:
We really would like to handle cascading updates on the server, but will leave that for a later task. Such a task has been created in Asana.